### ISSUE TITLE
"noexcept" C/C++ portability test doesn't really test what it should? 

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#28800]
AFAICT, the "noexcept" setting isn't really being propagated to the actual C++ cmake build, which makes the noexcept portability test useless?

https://github.com/grpc/grpc/blob/9ffd1a7b0a240e65534942dcabf4d3828d0c6be1/tools/run_tests/run_tests_matrix.py#L317

The cmdline
```
run_tests_c++_linux_noexcept_native: "python3 tools/run_tests/run_tests.py --use_docker -t -j 2 -x run_tests/cpp_linux_noexcept_native/sponge_log.xml --report_suite_name cpp_linux_noexcept_native -l c++ -c noexcept --iomgr_platform native"
```

It's unclear to me how the "-c noexcept" is propagated to the build and what compiler options it translates to.
This is probably a side effect of migrating from Makefile to cmake a long time ago.

**Comment by veblush** [comment]
Yeah I think this is a side effect of cmake migration. https://github.com/grpc/grpc/blob/9f1663fb74e6f5977f6b4bf1ad8b8005ac8616d3/build_handwritten.yaml#L132 still has a configuration for this build so we can do the same thing with cmake.

**Comment by yashykt** [comment]
Should this be on buganizer?
